packages/eslint-plugin/src/rules/no-shorthands.ts

```diff
@@ -77,6 +77,64 @@ const UNSUPPORTED_CSS_PROPERTIES: Record<keyof CSS.StandardShorthandProperties,
   transition: true,
 };
 
+type ShorthandToArguments = (value: string | number) => string[] | null;
+
+// Matches CSS functions, i.e. `rgb(255, 255, 255)` or `var(--color)`, as well as CSS values, i.e. `1px`, `solid`, and `red`.
+const CSS_FUNCTION_OR_VALUE = /(?:[a-zA-Z]+\(.*\)|[#\w-]+)/g;
+
+/**
+ * Splits a string into an array of CSS functions and values.
+ * If a number is provided, we assume it represents a pixel value.
+ */
+const splitShorthandIntoParts: ShorthandToArguments = value => {
+  if (typeof value !== 'string') {
+    if (value === 0) {
+      value = '0';
+    } else {
+      value = `${value}px`;
+    }
+  }
+  return value.match(CSS_FUNCTION_OR_VALUE);
+};
+
+// Transforms shorthand string into args for griffel shorthands.<name>() function.
+const SHORTHAND_FUNCTIONS: Partial<Record<keyof CSS.StandardShorthandProperties, ShorthandToArguments>> = {
```

Comment:
```suggestion
const SHORTHAND_FUNCTIONS: Partial<Record<keyof typeof UNSUPPORTED_CSS_PROPERTIES, ShorthandToArguments>> = {
```

nit: let's reuse the type that we already have